### PR TITLE
Adds support for opting out of telemetry via the DO_NOT_TRACK envirnment variable

### LINKED ISSUE #7366: Does not inform sufficiently of intent to report my data to Google Analytics
The netdata installer says the following:

```
  The following changes will be made to your system:

  # USERS / GROUPS
  User 'netdata' and group 'netdata' will be added, if not present.

  # LOGROTATE
  This file will be installed if logrotate is present.

   - /etc/logrotate.d/netdata

  # SYSTEM INIT
  This file will be installed if this system runs with systemd:

   - /lib/systemd/system/netdata.service

   or, for older Centos, Debian/Ubuntu or OpenRC Gentoo:

   - /etc/init.d/netdata         will be created


  This package can also update a netdata installation that has been
  created with another version of it.

  Your netdata configuration will be retained.
  After installation, netdata will be (re-)started.

  netdata re-distributes a lot of open source software components.
  Check its full license at:
  https://github.com/netdata/netdata/blob/master/LICENSE.md
Please type y to accept, n otherwise:
```

However, it does not tell me that it's going to silently spy on me.  I don't consent to analytics.  It spied on me and sent my data away without my consent.  I had to uninstall it, clean my system, and reinstall it using an obscure, barely-documented flag.

This is not respecting user consent.  Notify the user before spying on them, if you intend to do so, so they have an opportunity to cancel and delete your software if they don't wish for that to happen.

### LINKED ISSUE #7401: Inform users about telemetry and how to opt-out
Indicate in all installers before the Enter/yes-no: "Anonymous statistics will be [enabled/disabled]", according to the provided option. Provide information on the option relevant to the specific installer, to opt-out:

- [x] netdata-installer.sh
- [x] kickstart.sh 
- [x] kickstart-static64.sh
- [ ] binary packages (not sure about this one)

For docker images:

- [x] Add a `DO_NOT_TRACK` env var which when set will disable the anonymous stat pings.
- [x] Document the env var

Related: https://github.com/netdata/netdata/issues/7404#event-2848554127

### LINKED ISSUE #7404: Allow opt-out of telemetry via environment variables and command-line args
See https://github.com/netdata/netdata/issues/7366#issuecomment-560169517 from @titpetric 
We need to make it simpler for people to opt-out from telemetry when installing via docker (and obviously document the fact) and add a CLI option to netdata. 

Related: https://github.com/netdata/netdata/issues/7401

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

As per title

- Fixes #7366
- Fixes #7401
- Fixes #7404

##### Component Name

area/packing

##### Additional Information

- Depends on #7847

### CODE REVIEW

**Comment by ilyam8** [comment]
@prologic it is better to have code formatting changes in a separate PR or at least in a separate commit

**Comment by prologic** [comment]
> @prologic it is better to have code formatting changes in a separate PR or at least in a separate commit

Yes I agree; Please review #7847 first; I can rebase on top of that. I'll have to remember to do this before I touch anything -- [vim-autoformat](https://github.com/Chiel92/vim-autoformat) is pretty magical ;)

File: netdata-installer.sh [hunk 1]
```diff
@@ -698,226 +698,230 @@ run find "${NETDATA_WEB_DIR}" -type d -exec chmod 0775 {} \;
...
+    rm -rf "${tmp}"
+  fi
+  return 0
 }
 install_go
 
 # -----------------------------------------------------------------------------
 progress "Telemetry configuration"
 
+if [ x"$DO_NOT_TRACK" = x"1" ]; then
```
Comment (knatsakis) [inline netdata-installer.sh]: Why not simply `if [ "${DO_NOT_TRACK}" = "1" ]; then` ?

```suggestion
if [ "$DO_NOT_TRACK" = "1" ]; then
```

File: packaging/installer/kickstart-static64.sh [hunk 2]
```diff
@@ -188,64 +188,68 @@ NETDATA_INSTALLER_OPTIONS=""
...
+
+    NETDATA_LOCAL_TARBALL_OVERRIDE_CHECKSUM="${1}"
+    shift 1
+  else
+    echo >&2 "Unknown option '${1}' or invalid number of arguments. Please check the README for the available arguments of ${0} and try again"
+    exit 1
+  fi
 done
 
+if [ x"$DO_NOT_TRACK" = x"1" ]; then
```
Comment (knatsakis) [inline packaging/installer/kickstart-static64.sh]: Same as above. I have seen this pattern used elsewhere, but I don't think I understand the reasoning behind it. Could you explain a bit ?

```suggestion
if [ "$DO_NOT_TRACK" = "1" ]; then
```

File: packaging/makeself/install-or-update.sh [hunk 3]
```diff
@@ -29,75 +29,75 @@ REINSTALL_OPTIONS=""
...
-deleted_stock_configs=0
-if [ ! -f "etc/netdata/.installer-cleanup-of-stock-configs-done" ]
-then
-
-    # -----------------------------------------------------------------------------
-    progress "Deleting stock configuration files from user configuration directory"
-
-    declare -A configs_signatures=()
-    source "system/configs.signatures"
+if [ x"$DO_NOT_TRACK" = x"1" ]; then
```
Comment (knatsakis) [inline packaging/makeself/install-or-update.sh]: Same as above

```suggestion
if [ "$DO_NOT_TRACK" = "1" ]; then
```

**Comment by prologic** [comment]
> I am approving but what about the shellcheck warnings?
> 
> Oh, I suppose those are fixed by #7847?

Yes :D -- I created the PR to clean / fix what I was touching _after_ the fact. Sorry about that! Once #7847 is merged I"ll rebase this!

Also FYI but the form `if [ x"$foo" = x"bar" ]; then` is _actually_ the correct way to handle the edge-case where `$foo` can be blank. See this post: https://stackoverflow.com/questions/3093472/why-x-used-in-xvar-xvalue

I _think_ this has a lot to do with some badly written Shells and the fact that on most systems `[` and `]` are actually separate programs/binaries you can find in `/bin` normally.

**Comment by prologic** [comment]
Rebased now #7847 is merged.

File: packaging/makeself/install-or-update.sh [hunk 4]
```diff
@@ -57,6 +57,10 @@ while [ "${1}" ]; do
   shift 1
 done
 
+if [ x"$DO_NOT_TRACK" = x"1" ]; then
```
Comment (ncmans) [inline packaging/makeself/install-or-update.sh]: Oh I just remembered that we only need `DO_NOT_TRACK` set to a non-empty string, not to any particular value. See below:

https://github.com/netdata/netdata/blob/fa92c1e3b4990d76a87bb6d85297c954b667f397/packaging/docker/Dockerfile#L66

https://github.com/netdata/netdata/blob/fa92c1e3b4990d76a87bb6d85297c954b667f397/packaging/docker/run.sh#L10

https://github.com/netdata/netdata/pull/7437#issuecomment-561937299

Comment (prologic) [reply packaging/makeself/install-or-update.sh]: I _read_ this as `export DO_NOT_TRACK=1` elsewhere on the Internet 🤣

Wo which is it? :)

Which should we support? One form, both? I opted for making it consistently `DO_NOT_TRACK=1` everywhere now.

Comment (ncmans) [reply packaging/makeself/install-or-update.sh]: This is done by one dude and it's not really a standard or anything, so I'm not fussed which one. The [comment I linked](https://github.com/netdata/netdata/pull/7437#issuecomment-561937299) is the author himself saying it should work with any value.

Comment (prologic) [reply packaging/makeself/install-or-update.sh]: Ahh okay; I'll buy it. Give me a few mins to amend the PRs :)

Comment (prologic) [reply packaging/makeself/install-or-update.sh]: Done

**Comment by prologic** [comment]
Tested that this works as expected:

```sh
/netdata # find / -type f -name '.opt-out-from-anonymous-statistics'
/etc/netdata/.opt-out-from-anonymous-statistics
/netdata # grep XXX /var/log/netdata/*
/var/log/netdata/error.log:2020-01-29 00:16:09: netdata INFO  : MAIN : XXX: anonymous statistics disabled
```

File: docs/anonymous-statistics.md [hunk 5]
```diff
@@ -78,7 +78,8 @@ installation, including manual, offline, and macOS installations. Create the fil
 
 **Pass the option `--disable-telemetry` to any of the installer scripts in the [installation
 docs](../packaging/installer/README.md).** You can append this option during the initial installation or a manual
-update.
+update. You can also export the environment variable `DO_NOT_TRACK` with a non-zero or non-emppty value
```
Comment (ncmans) [inline docs/anonymous-statistics.md]: Typo:

```suggestion
update. You can also export the environment variable `DO_NOT_TRACK` with a non-zero or non-empty value
```

**Pull request merged**